pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/ModelParser.groovy

```diff
@@ -244,10 +244,22 @@ class ModelParser implements Parser {
                                 r.variables[key] = parseArgument(exp.rightExpression, true)
                                 return
                             } else if (exp.rightExpression instanceof MethodCallExpression) {
-                                r.variables[key] = parseInternalFunctionCall((MethodCallExpression)exp.rightExpression)
+                                r.variables[key] = parseInternalFunctionCall((MethodCallExpression) exp.rightExpression)
                                 return
+                            } else if (exp.rightExpression instanceof BinaryExpression) {
+                                if (((BinaryExpression)exp.rightExpression).operation.type  == Types.PLUS) {
+                                    // This is to support JENKINS-42771, allowing `FOO = "b" + "a" + "r"` sorts of syntax.
+                                    String envValue = envValueForStringConcat((BinaryExpression) exp.rightExpression)
```

Comment:
The latter will work - turns into `"4Bobby"`. The former...well, it'll be `"15"` currently. Need to think about it a bit - probably make sure at least one of the elements is a `String` or `GString`.